### PR TITLE
karmada-search: short circuit event hander for resourceRegistry updation

### PR DESCRIPTION
**Pull request by calvin0327**
Signed-off-by: calvin <wen.chen@daocloud.io>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
If  spec of resourceRegistry has not changed, we should not to registry reource informer  once again.  in the case, short circuit resourceRegistry update event.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
/assign @liys87x 

@liys87x By the way, we need an E2E test for `karmada-search` basic functionalities.

**Comment by liys87x** [comment]
> 

ok

**Comment by liys87x** [comment]
Thank you very much @calvin0327, I have tried it.

**Comment by calvin0327** [comment]
> Thank you very much @calvin0327, I will try it.

@liys87x thanks for review

File: pkg/search/controller.go [hunk 1]
```diff
@@ -264,11 +264,16 @@ func (c *Controller) updateResourceRegistry(oldObj, newObj interface{}) {
 	oldRR := oldObj.(*v1alpha1.ResourceRegistry)
 	newRR := newObj.(*v1alpha1.ResourceRegistry)
 
+	if reflect.DeepEqual(oldRR.Spec, newRR.Spec) {
```
Comment (liys87x) [inline pkg/search/controller.go:267]: make sense

Comment (calvin0327) [reply pkg/search/controller.go:267]: okay.

**Comment by calvin0327** [comment]
@RainbowMango PTAL

File: pkg/search/controller.go [hunk 2]
```diff
@@ -264,11 +264,16 @@ func (c *Controller) updateResourceRegistry(oldObj, newObj interface{}) {
...
+		klog.Infof("spec of resourceRegistry[%s] has not changed", newRR.Name)
```
Comment (RainbowMango) [inline pkg/search/controller.go:268]:
```suggestion
		klog.V(4).Infof("Ignore ResourceRegistry(%s) update event as spec not changed", newRR.Name) 
```

This log can be suppressed or ignored I think. I can't see why we need it.

Comment (calvin0327) [reply pkg/search/controller.go]: @RainbowMango That's right.  the update event message doesn't make sense. 
I hadn't thought of that，it's good.
Done

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**